compiler-core/src/format/tests.rs

```diff
@@ -18,20 +18,36 @@ mod use_;
 #[macro_export]
 macro_rules! assert_format {
     ($src:expr $(,)?) => {
-        let mut writer = String::new();
-        $crate::format::pretty(&mut writer, &$src.into(), camino::Utf8Path::new("<stdin>"))
-            .unwrap();
-        assert_eq!($src, writer);
+        $crate::assert_format_rewrite!($src, $src);
     };
 }
 
 #[macro_export]
 macro_rules! assert_format_rewrite {
     ($src:expr, $output:expr  $(,)?) => {
-        let mut writer = String::new();
-        $crate::format::pretty(&mut writer, &$src.into(), camino::Utf8Path::new("<stdin>"))
-            .unwrap();
-        assert_eq!(writer, $output);
+        // Different EOLs in input should all format to \n in the output
+
+        // \n
+        {
+            let mut writer = String::new();
+            let src = $src.into();
+            $crate::format::pretty(&mut writer, &src, camino::Utf8Path::new("<stdin>")).unwrap();
+            assert_eq!(writer, $output);
+        }
+        // \r\n
+        {
+            let mut writer = String::new();
+            let src = $src.replace('\n', "\r\n").into();
+            $crate::format::pretty(&mut writer, &src, camino::Utf8Path::new("<stdin>")).unwrap();
+            assert_eq!(writer, $output);
+        }
+        // \r
+        {
+            let mut writer = String::new();
+            let src = $src.replace('\n', "\r").into();
+            $crate::format::pretty(&mut writer, &src, camino::Utf8Path::new("<stdin>")).unwrap();
+            assert_eq!(writer, $output);
+        }
```

Comment:
Let's write a couple new tests instead of running the whole test suite 3 times.

Reply:
Fair point. How about a third macro that tests with all the eol variations, and use that for selected test cases? Mainly those which check the empty line formatting.

Would that be good?

And if, would it be better as separate macros e.i. `assert_format_rewrite_with_eol_conventions`, or with extra argument (and same for `assert_format`):
```rs
#[macro_export]
macro_rules! assert_format_rewrite {
    ($src:expr, $output:expr  $(,)?) => { /* main impl */ };

    (check_eol_conventions, $src:expr, $output:expr  $(,)?) => {
        // Different EOLs in input should all format to \n in the output
        $crate::assert_format_rewrite!($src, $output);
        $crate::assert_format_rewrite!(&$src.replace('\n', "\r\n"), $output);
        $crate::assert_format_rewrite!(&$src.replace('\n', "\r"), $output);
    };
}
```

Reply:
I'd prefer to keep it simple and write three tests in this case.

Reply:
I somewhat disagree about this being a one-off. I might even consider this being a general property of formatting: eol normalisation should not change the end result.

The trailing white space case is not quite as general, since comments (at least currently) retain the space.